homeassistant/components/workday/repairs.py

```diff
@@ -0,0 +1,175 @@
+            self.hass.config_entries.async_update_entry(self.entry, options=new_options)
+            await self.hass.config_entries.async_reload(self.entry.entry_id)
+            return self.async_create_entry(data={})
+
+        assert self.country
+        country_provinces = list_supported_countries()[self.country]
+        return self.async_show_form(
+            step_id="province",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONF_PROVINCE, default=NONE_SENTINEL): SelectSelector(
+                        SelectSelectorConfig(
+                            options=[NONE_SENTINEL, *country_provinces],
+                            mode=SelectSelectorMode.DROPDOWN,
+                            translation_key=CONF_PROVINCE,
+                        )
+                    ),
+                }
+            ),
+            description_placeholders={
+                CONF_COUNTRY: self.country,
+                "title": self.entry.title,
+            },
+        )
+
+
+class ProvinceFixFlow(RepairsFlow):
+    """Handler for an issue fixing flow."""
+
+    def __init__(self, entry: ConfigEntry) -> None:
+        """Create flow."""
+        self.entry = entry
+        super().__init__()
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the first step of a fix flow."""
+
+        return await self.async_step_province()
+
+    async def async_step_province(
+        self, user_input: dict[str, Any] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the province step of a fix flow."""
+        if user_input and user_input.get(CONF_PROVINCE):
+            if user_input.get(CONF_PROVINCE, NONE_SENTINEL) == NONE_SENTINEL:
+                user_input[CONF_PROVINCE] = None
+            options = dict(self.entry.options)
+            new_options = {**options, **user_input}
+            self.hass.config_entries.async_update_entry(self.entry, options=new_options)
+            await self.hass.config_entries.async_reload(self.entry.entry_id)
+            return self.async_create_entry(data={})
+
+        country = self.entry.options[CONF_COUNTRY]
+        all_countries = list_supported_countries()
+        province_list = [NONE_SENTINEL, *all_countries[country]]
+        return self.async_show_form(
+            step_id="province",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONF_PROVINCE, default=NONE_SENTINEL): SelectSelector(
+                        SelectSelectorConfig(
+                            options=province_list,
+                            mode=SelectSelectorMode.DROPDOWN,
+                            translation_key=CONF_PROVINCE,
+                        )
+                    ),
+                }
+            ),
+            description_placeholders={CONF_COUNTRY: country, "title": self.entry.title},
+        )
```

Comment:
This is almost exactly the same code as in `CountryFixFlow.async_step_province`, please refactor to avoid duplication.

One way would be to use a single class, and when fixing the province create a `CountryFixFlow` and set the `country` based on config entry data.